src/Illuminate/Support/Carbon.php

```diff
@@ -30,8 +20,10 @@ public static function setTestNow($testNow = null)
      */
     public static function createFromId($id)
     {
-        return Ulid::isValid($id)
-            ? static::createFromInterface(Ulid::fromString($id)->getDateTime())
-            : static::createFromInterface(Uuid::fromString($id)->getDateTime());
+        if (\is_string($id)) {
+            $id = Ulid::isValid($id) ? Ulid::fromString($id) : Uuid::fromString($id);
+        }
+
+        return static::createFromInterface($id->getDateTime());
```

Comment:
> Noticing createFromId PHPDoc says it accepts Uuid and Ulid but code fails if not string so I changed it to respect its PHPDoc

Doesn't the same code be applied to Laravel 10 and Carbon 2.x?

Reply:
Yes I can make a separated PR for that.

Reply:
Extracted: https://github.com/laravel/framework/pull/49783